layers.py

```diff
@@ -0,0 +1,67 @@
+import tensorflow as tf
+import tensorflow.keras as keras
+from tensorflow.keras.layers import *
+
+def high_level_feature_representation_base(config):
+    t_pool_size = [int(i) for i in config.high_level_feature_representation_pool1.split(',')]
+    pool_size = [int(i) for i in config.high_level_feature_representation_pool2.split(',')]
+    
+    def _high_level_feature_representation(inputs):
+        x = inputs
+        for i, convCnt in enumerate(pool_size):
+            x = Conv2D(config.high_level_feature_representation_filter, kernel_size=3, padding='same')(
+                x)
+            x = BatchNormalization()(x)
+            x = Activation('relu')(x)
+            x = MaxPooling2D(pool_size=(t_pool_size[i], pool_size[i]))(
+                x)
+            x = Dropout(config.dropout_rate)(x)
+        outputs = x
+        return outputs
+    return _high_level_feature_representation
+
+
+def temporal_context_representation_base(config):
+    rnn_size = [int(i) for i in config.temporal_context_representation_filter.split(',')]
+
+    def _temporal_context_representation(inputs):
+        x = inputs
+        # [b, t, f, c] -> [b, t, c]
+        x = Reshape((-1, x.shape[-2]*x.shape[-1]))(x)
+        for nb_rnn_filt in rnn_size:
+            x = Bidirectional(
+                GRU(nb_rnn_filt, activation='tanh', 
+                    dropout=config.dropout_rate, recurrent_dropout=config.dropout_rate, 
+                    return_sequences=True),
+                merge_mode='mul')(x)
+        outputs = x
+        return outputs
+    return _temporal_context_representation
+
+
+def sed_layer_base(config, n_classes):
+    fnn_size = [int(i) for i in config.sed_layer_filter.split(',')]
+    
+    def _sed_layer(inputs):
+        x = inputs
+        for nb_fnn_filt in fnn_size:
+            x = TimeDistributed(Dense(nb_fnn_filt))(x)
+            x = Dropout(config.dropout_rate)(x)
+        x = TimeDistributed(Dense(n_classes, name='sed_out'))(x)
+        outputs = x
+        return outputs
+    return _sed_layer
+
+
+def doa_layer_base(config, n_classes):
```

Comment:
if config is given, why additional "n_classes" is required? how about removing n_classes? sed_layer_base also requires n_classes. it would be better to remove n_classes

Reply:
there is no class number in config, so I made it independently